torch/onnx/_internal/diagnostics/infra/formatter.py

```diff
@@ -0,0 +1,47 @@
+import dataclasses
+import json
+import re
+from typing import Any, Callable, Dict, List, Union
+
+from torch.onnx._internal.diagnostics.infra import sarif
+
+# A list of types in the SARIF module to support pretty printing.
+# This is solely for type annotation for the functions below.
+_SarifClass = Union[
+    sarif.SarifLog,
+    sarif.Run,
+    sarif.ReportingDescriptor,
+    sarif.Result,
+]
+
+
+def _camel_case_to_snake_case(s: str) -> str:
+    return re.sub(r"([A-Z])", r"_\1", s).lower()
+
+
+def _kebab_case_to_snake_case(s: str) -> str:
+    return s.replace("-", "_")
+
+
+def _convert_key(
+    object: Union[Dict[str, Any], Any], convert: Callable[[str], str]
+) -> Union[Dict[str, Any], Any]:
```

Comment:
To me what this function does is not very clear (or why/when one needs to use it). Consider a brief docstring?

Reply:
I can see now from the usage in `sarif_to_json`. Still, just a sentence should suffice.

Reply:
added some docstring